app/models/report.rb

```diff
@@ -4,32 +4,41 @@ class Report < ApplicationRecord
   validates :user_id, presence: true
   validates :item_id, presence: true
   validates :item_type, presence: true, inclusion: {
-    in: %w(Post Comment), message: "Type should match `Post` or `Comment`!"}
+    in: %w[Post Comment], message: "Type should match `Post` or `Comment`!"
+  }
   validates :text, presence: true
 
-  validate :entry_does_not_exist, :on => :create
-  validate :post_or_comment_does_exist, :on => :create
+  validate :entry_does_not_exist, on: :create
+  validate :post_or_comment_does_exist, on: :create
 
   belongs_to :user
   belongs_to :post, optional: true
   belongs_to :comment, optional: true
   belongs_to :item, polymorphic: true
 
-  after_commit :send_report_notification, :on => :create
+  STATUS_DELETED = "deleted"
+  STATUS_NO_ACTION = "no action"
```

Comment:
'no action' seems fine to me in that context. It makes clear that a decision has been made to take no action on that content. The label 'ignored' is potentially confusing: it could mean that the person who looked at the report decided to ignore the report (didn't make a decision), or that the content was 'ignored' i.e. hidden, or something else.

I'm not sure why some of the reports that have 'no action' next to them also have eye and bin icons to the right of them and others don't – is that clear to anyone using this feature?

Reply:
As Fla found out, with introduction of the new "action" field, no existing Reports have a default initial value. I've added a migration and set all existing Reports to 'NO_ACTION', so every reviewed report should now have the action buttons.